### PR TITLE
[Compute] BREAKING CHANGE: `az vm/vmss create`: Support Trusted Launch as default when creating vm/vmss with minimal inputs

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

close: #23440

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -412,8 +412,12 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
         else:
             raise CLIError('Unrecognized image type: {}'.format(image_type))
     else:
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:414]: Is it possible for a user to input both `--image` and `--attach-os-disk`? If so, please shrink the range of condition to not inputing neither `--image` nor `--attach-os-disk`.
```suggestion
    elif not namespace.image and not getattr(namespace, 'attach_os_disk', None):
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -412,8 +412,12 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
...
-        # did not specify image XOR attach-os-disk
-        raise CLIError('incorrect usage: --image IMAGE | --attach-os-disk DISK')
+        namespace.image = 'MicrosoftWindowsServer:WindowsServer:2022-datacenter-azure-edition:latest'
+        _parse_image_argument(cmd, namespace)
+        namespace.storage_profile = StorageProfile.ManagedPirImage
+        namespace.enable_secure_boot = True
+        namespace.enable_vtpm = True
+        namespace.security_type = 'TrustedLaunch'
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:420]: May I ask if these parameters are set to default value only in the absence of user input?
Hi @AjKundnani, could you please help confirm it?
```suggestion
        if namespace.enable_secure_boot is None:
            namespace.enable_secure_boot = True
        if namespace.enable_vtpm is None:
            namespace.enable_vtpm = True
        if namespace.security_type is None:
            namespace.security_type = 'TrustedLaunch'
```

**Comment by zhoxing-ms** [comment]
Replaced by https://github.com/Azure/azure-cli/pull/27749

**Pull request closed without merging**